traitlets/traitlets.py

```diff
@@ -649,14 +678,23 @@ def observe(*names):
     """
     return ObserveHandler(names)
 
+def validate(name):
+    """ A decorator which validates a HasTraits object's state when a member is set.
```

Comment:
It's only used when a Trait is set, right?  A HasTraits often has non-Trait members, and those don't use this at all.